### ISSUE TITLE
Add packaging tests using yum and apt

### ISSUE THREAD

**Issue by rjernst** [elastic/elasticsearch#39494]
While testing the yum and apt repositories is something outside the scope of a local elasticsearch test, both those tools support installing a local rpm and deb file, respectively. We should have packaging tests which ensure those tools can be used for installing and upgrading elasticsearch with a local package.

**Comment by mark-vieira** [comment]
@rjernst is this still an issue? How are we installing the packages for the current pckaging tests, just via `dpkg -i` (and the RPM equivalent)?

**Comment by rjernst** [comment]
Yes, I think it is still something we should test. We only use the package tools directly, but do not test package managers with our package files.

**Comment by mark-vieira** [comment]
Do we have any examples of scenarios in which there would be a difference? Or instances if which we introduced a regression related specifically to differences in the two?

**Comment by rjernst** [comment]
See the discussion and linked issues on https://github.com/elastic/elasticsearch/pull/39472.

**Comment by mark-vieira** [comment]
Thanks Ryan. I completely neglected to look at that issue :facepalm: We'll keep this issue around.

**Comment by pugnascotia** [comment]
@rjernst do you still think we need this?

**Comment by rjernst** [comment]
Yes, I think this is still needed.
